Gems/Atom/Tools/AtomToolsFramework/Code/Source/Graph/GraphDocument.cpp

```diff
@@ -574,5 +577,91 @@ namespace AtomToolsFramework
         }
 
         AtomToolsDocumentNotificationBus::Event(m_toolId, &AtomToolsDocumentNotificationBus::Events::OnDocumentObjectInfoInvalidated, m_id);
-    };
+    }
+
+    DocumentObjectInfoVector GraphDocument::GetObjectInfoForGraphCanvasNodes() const
+    {
+        DocumentObjectInfoVector objects;
+
+        // Reserve and register reflected objects for all of the selected graph canvas nodes that do not mirror any of the graph model nodes
+        // that have been added to the graph. This should cover bookmarks, comments, and groups.
+        AZStd::vector<AZ::EntityId> selectedItems;
+        GraphCanvas::SceneRequestBus::EventResult(selectedItems, m_graphId, &GraphCanvas::SceneRequests::GetSelectedItems);
+
+        // Optimizing the container to only have nodes with property components before sorting.
+        AZStd::erase_if(
+            selectedItems,
+            [](const auto& selectedItem)
+            {
+                return GraphCanvas::GraphCanvasPropertyBus::FindFirstHandler(selectedItem) == nullptr;
+            });
+
+        objects.reserve(objects.size() + selectedItems.size());
+
+        // The order that selected nodes appear in the container is not deterministic. To compensate for this, we sort by position to ensure
+        // that nodes always appear in the inspector in a consistent order.
+        AZStd::sort(
+            selectedItems.begin(),
+            selectedItems.end(),
+            [](const auto& selectedItem1, const auto& selectedItem2)
+            {
+                AZ::Vector2 selectedItemPosition1{};
+                GraphCanvas::GeometryRequestBus::EventResult(
+                    selectedItemPosition1, selectedItem1, &GraphCanvas::GeometryRequests::GetPosition);
+
+                AZ::Vector2 selectedItemPosition2{};
+                GraphCanvas::GeometryRequestBus::EventResult(
+                    selectedItemPosition2, selectedItem2, &GraphCanvas::GeometryRequests::GetPosition);
+
+                return selectedItemPosition1.IsLessThan(selectedItemPosition2);
+            });
+
+        // Some graph canvas node property components do not have any visible properties, like the bookmark anchor visual component. These
+        // will not be added to the graph document inspector.
+        const AZStd::unordered_set<AZ::Uuid> ignoredTypeIds{
+            AZ::Uuid::CreateString("{AD921E77-962B-417F-88FB-500FA679DFDF}") // BookmarkAnchorVisualComponent
```

Comment:
nitpick: The `AZ::Uuid` constructor can be used directly here instead of `CreateString`
```suggestion
            AZ::Uuid("{AD921E77-962B-417F-88FB-500FA679DFDF}") // BookmarkAnchorVisualComponent
```